packages/pwa-kit-dev/src/ssr/server/build-dev-server.js

```diff
@@ -152,6 +152,15 @@ export const DevServerMixin = {
         if (fs.existsSync(projectWebpackPath)) {
             config = require(projectWebpackPath)
         }
+
+        const clientConfig = config.find((cnf) => cnf.name === CLIENT)
+
+        // We update HMR basePath on the clientConfig here since it needs to happen before HMR is started
+        if (clientConfig && process.env.HMR !== 'false') {
+            const hmrBasePath = `${getBundlePath()}/development/`
+            clientConfig.output.publicPath = hmrBasePath
+        }
```

Comment:
Why wouldn't this be don't in the webpack config "enhancement" function "enableReactRefresh"? Seems like that would be the logical place to do that instead of spreading it across multiple files.

Reply:
Moved to webpack config `enableReactRefresh`.

(I was a bit hesitant to run `getBundlePath()` in webpack config, because it's meant to run at runtime not build time, but I tested it and it seems it's working properly, so 👍  )